core/blockchain_test.go

```diff
@@ -895,9 +908,10 @@ func TestChainTxReorgs(t *testing.T) {
 	})
 	// Import the chain. This runs all block validation rules.
 	evmux := &event.TypeMux{}
-	chainman, _ := NewBlockChain(db, FakePow{}, evmux)
-	chainman.SetProcessor(NewBlockProcessor(db, FakePow{}, chainman, evmux))
-	if i, err := chainman.InsertChain(chain); err != nil {
+	blockchain, _ := NewBlockChain(db, FakePow{}, evmux)
+	blockchain.SetValidator(NewBlockValidator(blockchain, FakePow{}))
+	blockchain.SetProcessor(NewStateProcessor(blockchain))
```

Comment:
AFAIK, these Setters are not needed any more. `NewBlockChain` should already initialize these internally.